cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -809,27 +818,141 @@ func (m *gceManagerImpl) getTemplates() *templateBuilder {
 }
 
 func (m *gceManagerImpl) Refresh() error {
-	if m.mode == ModeGCE {
+	if m.lastRefresh.Add(refreshInterval).After(time.Now()) {
 		return nil
 	}
-	if m.lastRefresh.Add(refreshInterval).Before(time.Now()) {
-		err := m.fetchAllNodePools()
-		if err != nil {
+	return m.forceRefresh()
+}
+
+func (m *gceManagerImpl) forceRefresh() error {
+	nextRefreshAfter := time.Now().Add(refreshInterval)
+	switch m.mode {
+	case ModeGCE:
+		if err := m.fetchAutoMigs(); err != nil {
+			glog.Errorf("Failed to fetch MIGs: %v", err)
 			return err
 		}
+		glog.V(2).Infof("Refreshed MIG list, next refresh after %v", nextRefreshAfter)
+	case ModeGKENAP:
+		if err := m.fetchResourceLimiter(); err != nil {
+			glog.Errorf("Failed to fetch resource limits: %v", err)
+			return err
+		}
+		glog.V(2).Infof("Refreshed resource limits, next refresh after %v", nextRefreshAfter)
+		fallthrough
+	case ModeGKE:
+		if err := m.fetchAllNodePools(); err != nil {
+			glog.Errorf("Failed to fetch node pools: %v", err)
+			return err
+		}
+		glog.V(2).Infof("Refreshed NodePool list, next refresh after %v", nextRefreshAfter)
+	}
+	m.lastRefresh = time.Now()
```

Comment:
I don't like logging one timestamp and than actually using a different one - it will be really confusing to someone debugging this in the future. We're doing multiple API calls in the meantime, so the difference is likely to be noticeable.

Maybe we can just log something like "Finished refreshing config, next refresh after" at the end? We can still keep old log messages, just without the "next refresh after" part.

Reply:
I figured this wasn't too bad since technically it said the next refresh would happen "after" the time, not "at" the time. ;) Fixed anyhow.